### ISSUE TITLE
vagrant up laravel/homestead failed with error: address family must be specified (IPAddr::AddressFamilyError)

### ISSUE THREAD

**Issue by viewplatgh** [hashicorp/vagrant#13144]
### Debug output

https://gist.github.com/viewplatgh/b5c9a7caa464d17cf195f78099ca11fb

### Expected behavior

homestead virtualbox vm should be started successfully

### Actual behavior

homestead virtualbox vm failed to start with the error as per in debug ouput:
/opt/vagrant/embedded/gems/2.3.4/gems/ipaddr-1.2.5/lib/ipaddr.rb:594:in `initialize': address family must be specified (IPAddr::AddressFamilyError)

### Reproduction information 
follow laravel/homestead instruction at: https://laravel.com/docs/8.x/homestead

#### Vagrant version
2.3.4

#### Host operating system
macos catalina 10.15.7
iMac13,2

#### Guest operating system
laravel/homestead v14.3.1 Ubuntu LTS 20.04

#### Steps to reproduce

1. install virtualbox 6.1.44
2. install vagrant 2.3.4
3. git clone homestead and checkout release branch
4. init Homestead configuration: bash init
5. configure Homestead.yaml settings: folders, sites
6. run: vagrant up

#### Vagrantfile
```
# -*- mode: ruby -*-
# vi: set ft=ruby :

require 'json'
require 'yaml'

VAGRANTFILE_API_VERSION ||= "2"
confDir = $confDir ||= File.expand_path(File.dirname(__FILE__))

homesteadYamlPath = confDir + "/Homestead.yaml"
homesteadJsonPath = confDir + "/Homestead.json"
afterScriptPath = confDir + "/after.sh"
customizationScriptPath = confDir + "/user-customizations.sh"
aliasesPath = confDir + "/aliases"

require File.expand_path(File.dirname(__FILE__) + '/scripts/homestead.rb')

Vagrant.require_version '>= 2.2.4'

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
    if File.exist? aliasesPath then
        config.vm.provision "file", source: aliasesPath, destination: "/tmp/bash_aliases"
        config.vm.provision "handle_aliases", type: "shell" do |s|
            s.inline = "awk '{ sub(\"\r$\", \"\"); print }' /tmp/bash_aliases > /home/vagrant/.bash_aliases && chown vagrant:vagrant /home/vagrant/.bash_aliases"
        end
    end

    if File.exist? homesteadYamlPath then
        settings = YAML::load(File.read(homesteadYamlPath))
    elsif File.exist? homesteadJsonPath then
        settings = JSON::parse(File.read(homesteadJsonPath))
    else
        abort "Homestead settings file not found in #{confDir}"
    end

    Homestead.configure(config, settings)

    if File.exist? afterScriptPath then
        config.vm.provision "Run after.sh", type: "shell", path: afterScriptPath, privileged: false, keep_color: true
    end

    if File.exist? customizationScriptPath then
        config.vm.provision "Run customize script", type: "shell", path: customizationScriptPath, privileged: false, keep_color: true
    end

    if Vagrant.has_plugin?('vagrant-hostsupdater')
        config.hostsupdater.remove_on_suspend = false
        config.hostsupdater.aliases = settings['sites'].map { |site| site['map'] }
    elsif Vagrant.has_plugin?('vagrant-hostmanager')
        config.hostmanager.enabled = true
        config.hostmanager.manage_host = true
        config.hostmanager.aliases = settings['sites'].map { |site| site['map'] }
    elsif Vagrant.has_plugin?('vagrant-goodhosts')
        config.goodhosts.aliases = settings['sites'].map { |site| site['map'] }
    end

    if Vagrant.has_plugin?('vagrant-notify-forwarder')
        config.notify_forwarder.enable = true
    end
end
```

**Comment by guoguicheng** [comment]
- Upgrade virtualbox v7.0.8
- Upgrade homestead version
`brew cleanup`
`brew update`
`brew reinstall --cask vagrant`
- Restart virtualbox
`sudo "/Library/Application Support/VirtualBox/LaunchDaemons/VirtualBoxStartup.sh" restart`
- Restart Vagrant
`vagrant halt`
`vagrant up`

Hope it works for you!

**Comment by chrisroberts** [comment]
Hi there,

If you are still encountering this issue, please provide a minimal Vagrantfile that reproduces the error an I will be happy to investigate further.

Cheers!

**Comment by viewplatgh** [comment]
> 

Hi Chris,
The Vagrantifile I am using is: https://github.com/laravel/homestead/blob/main/Vagrantfile. Just follow the tutorial: https://laravel.com/docs/8.x/homestead, I can reproduce the issue. I remember I started getting this problem when I tried upgrading virtualbox to v7.x.x. Then I can't get rid of this issue no matter how, including downgrading virtualbox to v6.x.x. I feel this problem might be a very specific case due to my special local environment.
